provider/metadata_test.go

```diff
@@ -30,6 +30,7 @@ func TestMetadata(t *testing.T) {
 					item {
 						key = "foo"
 						value = "bar"
+						order = 4
```

Comment:
Can you add more test cases?

1. All items have `order` present.
2. Order is negative.
3. Only a few items have `order` present